erdos/cluster.py

```diff
@@ -0,0 +1,82 @@
+from pexpect import pxssh
+
+
+class Cluster(object):
+    """Cluster base class.
+
+    All clusters must inherent from this class, and must implement:
+
+    1. setup_node: Runs commands to connect the node to the cluster.
+    """
+
+    def __init__(self):
+        self.nodes = {}
+
+    def __del__(self):
+        for node in self.nodes.values():
+            node.disconnect()
+
+    def add_node(self,
+                 name,
+                 server,
+                 username,
+                 ssh_key,
+                 setup_env_commands=None):
+        """Adds a node to the cluster and sets it up"""
+        node = Node(name, server, username, ssh_key)
+        self.nodes[name] = node
+
+        if setup_env_commands is not None:
+            for command in setup_env_commands:
+                node.run_command(command)
+
+        self.setup_node(node)
+
+        return node
+
+    def setup_node(self, node):
```

Comment:
This should be called `setup` and be invoked to setup the cluster. Internally, the method may invoke `setup` on each node.